### PR TITLE
feat: Gamify authentication with firebase

### PR DESCRIPTION
**Pull request by sanderelderman**
New functionality:

1. Auth module with firebase authentication strategy and guard
2. Firebase user parameter decorator
3. Users module with service methods for finding, creating and updating the basics of the `User` model
4. Added config module to the core module for loading `.env` from core lib
5. Add role based authentication guard and apply globaly
6. Roles decorator for guarding routes behind certain roles
7. Apply role decorator to relevant guards in `Application` and `Badge` module

### CODE REVIEW

File: libs/gamify/auth/src/lib/decorators/roles.decorator.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { Role } from "@gamify/core";
+import { SetMetadata } from "@nestjs/common";
+
+export const Roles = (...roles: Role[]) => SetMetadata('roles', roles);
```
Comment (mschilling) [inline libs/gamify/auth/src/lib/decorators/roles.decorator.ts:4]: Nice to have this decorator

File: libs/gamify/auth/src/lib/guards/roles.guard.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { CanActivate, ExecutionContext, Injectable } from "@nestjs/common";
+import { Reflector } from "@nestjs/core";
+
+@Injectable()
+export class RolesGuard implements CanActivate {
+
+    constructor(private reflector: Reflector) {}
+
+    canActivate(context: ExecutionContext): boolean {
+        const roles = this.reflector.get<string[]>('roles', context.getHandler());
```
Comment (mschilling) [inline libs/gamify/auth/src/lib/guards/roles.guard.ts:10]: Future improvement: introduce a `MetaKeys` enum for this (or something similar).

File: libs/gamify/core/sample.env [hunk 3]
```diff
@@ -0,0 +1,4 @@
+GOOGLE_APPLICATION_CREDENTIALS=./path/to/service-account.json
```
Comment (mschilling) [inline libs/gamify/core/sample.env:1]: Good to have a sample.env

File: libs/gamify/users/src/lib/users.controller.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { Controller } from '@nestjs/common';
+import { UsersService } from './users.service';
+
+@Controller('users')
+export class UsersController {
+  constructor(private usersService: UsersService) {}
```
Comment (mschilling) [inline libs/gamify/users/src/lib/users.controller.ts:6]: Any thoughts on what you want to achieve here? Return all database users?

**Review by mschilling — APPROVED** [review]
It would be nice to add a 'Getting Started' or 'Development Setup' to a README in the app/gamify folder.

Other than this, it looks good. Great progress!

File: libs/gamify/users/src/lib/users.controller.ts (hunk 4 above)
Comment (sanderelderman) [reply libs/gamify/users/src/lib/users.controller.ts:6]: Yes, this controller will be used in future features including returning all users and assigning a badge to a user.

**Pull request merged**